pkg/metrics/metrics.go

```diff
@@ -50,6 +52,15 @@ func (mc *MetricPrometheusContext) ObserveRequest(err error) error {
 	return mc.Observe(apiRequestPrometheusMetrics, err)
 }
 
+// ObserveRequestIgnoreNotFound records the request latency and counts the errors if it's not IsNotFound.
+func (mc *MetricPrometheusContext) ObserveRequestIgnoreNotFound(err error) error {
+	if capoerrors.IsNotFound(err) {
+		_ = mc.ObserveRequest(nil)
+		return err
```

Comment:
this line return err (NotFound) here
but the function name seems saying the error will be ignored? did I miss something?

Reply:
`ObserveRequest` always returns the `err` and *observes* all metrics
`ObserveRequestIgnoreNotFound` always returns `err` and handles all metrics except for `IsNotFound`.

These functions are only for metrics, not for error handling :)

Reply:
Addition: the `IsNotFound` metrics will also be observed, but not as an error metric (i.e., `error` is `nil`)

Reply:
ok , got it ~ thanks for the detailed info

Reply:
You're welcome :)